README.md

```diff
@@ -0,0 +1,140 @@
+
+Nuxeo's ecosystem is Java based and uses Maven. This addon is not an exception and can be built by simply performing:
+
+```shell script
+mvn clean install
+```
+
+This will build all the modules except _ci_ and generate the correspondent artifacts: _`.jar`_ files for the contributions, and a _`.zip_ file for the package.
+
+### Frontend Contribution
+
+`nuxeo-coldstorage-web` module is also generating a _`.jar`_ file containing all the artifacts needed for an integration with Nuxeo's ecosystem.
+Nevertheless this contribution is basically generating an ES Module ready for being integrated with Nuxeo Web UI.
+
+It is possible to isolate this part of the build by running the following command:
+
+```shell script
+npm run build
+```
+
+It is using [rollup.js](https://rollupjs.org/guide/en/) to build, optimize and minify the code, making it ready for deployment.
+
+## Test
+
+In a similar way to what was written above about the building process, it is possible to run tests against each one of the modules.
+
+Here, despite being under the same ecosystem, the contributions use different approaches.
+
+### Backend Contribution
+
+#### Unit Tests
+
+```shell script
+mvn test
+```
+
+### Frontend Contribution
+
+#### Unit Tests
+
+```shell script
+npm run test
+```
+
+[Web Test Runner](https://modern-web.dev/docs/test-runner/overview/) is the test runner used to run this contribution unit tests.
+The tests run against bundled versions of Chromium, Firefox and Webkit, using [Playwright](https://www.npmjs.com/package/playwright)
+
+#### Functional Tests
+
+```shell script
+npm run ftest
+```
+
+To run the functional tests, [Nuxeo Web UI Functional Testing Framework](https://github.com/nuxeo/nuxeo-web-ui/tree/master/packages/nuxeo-web-ui-ftest) is used.
+Due to its inner dependencies, it only works using NodeJS `lts/dubnium`, i.e., `v10`.
+
+## Development Workflow
+
+### Frontend
+
+*Disclaimer:* In order to contribute and develop Nuxeo Cold Storage UI, it is assumed that there is a Nuxeo server running with Nuxeo Cold Storage package installed and properly configured according the documentation above.
+
+#### Install Dependencies  
+
+```sh
+npm install
+```
+
+#### Linting & Code Style
+
+The UI contribution has linting to help making the code simpler and safer.
+
+```sh
+npm run lint
+```
+
+To help on code style and formatting the following command is available. 
+
+```sh
+npm run format
+```
+
+Both `lint` and `format` commands run automatically before performing a commit in order to help us keeping the code base consistent with the rules defined.
+
+#### Integration with Web UI
+
+Despite being an "independent" project, this frontend contribution is build and aims to run as part of Nuxeo Web UI. So, most of the development will be done under that context.
+To have the best experience possible, it is recommended to follow the `Web UI Development workflow` on [repository's README](https://github.com/nuxeo/nuxeo-web-ui/blob/master/README.md).
+
```

Comment:
for web-ui we should point to the lts-2020 of web-ui, I think the maintenance branch is https://github.com/nuxeo/nuxeo-web-ui/tree/maintenance-3.0.x